### ISSUE TITLE
can't reproduce results for gradient, grad_times_input and integrated_gradients

### ISSUE THREAD

**Issue by laura-rieger** [pankessel/explanations_can_be_manipulated#1]
Hi, thanks for the insightful paper and making the code public! 
While trying to reproduce your results (namely figure 2 in the paper), I ran into issues with multiple attribution methods: 
For gradient and guided backprop, the explanation visually does not change at all.
For grad_times_input the explanation does not look like the target explanation and the image is visibly disturbed. 
For LRP the method works as expected, so I feel like it's not a general issue.
In all cases I used the parameters suggested in appendix A (num_iterations, learning_rate) and otherwise did not change your implementation or default parameters.
Do you know what could be the issue or are there other parameters that need to be changed? Thanks for your help!
Guided backprop:
![overview_guided_backprop](https://user-images.githubusercontent.com/10225326/72351723-2e447000-36e1-11ea-89e1-444cb67b6ac8.png)

Grad x input:
![overview_grad_times_input](https://user-images.githubusercontent.com/10225326/72351734-313f6080-36e1-11ea-8914-44ea5f604c8b.png)

**Comment by pankessel** [comment]
Thank you for your interest in our work and for contacting us. This is much appreciated!

For gradient-based explanation methods, you will want to use beta growth which can be activated using the --beta_growth flag in the run_attack script. Please have a look at Section A1 in our supplement for more details.

Hope this helps!

**Comment by laura-rieger** [comment]
Thanks for your answer! This would indeed fix the problem. However, looking at section A.1 it says that  beta growth is not essential for the results, merely beneficial. Even if I fix beta to be 10 or 100, I get similar results to the ones above. Do you know why this could be the case?

**Comment by pankessel** [comment]
Can you provide us with the precise arguments that you used to run the run_attack script (and the code if you modified it)? We used beta growth for our experiments but we checked that it only mildly improves the results. So this seems a bit odd...

**Comment by laura-rieger** [comment]
I have not changed anything in the code and ran

python  run_attack.py --cuda --method gradient  --lr 0.001

python  run_attack.py --cuda --method guided_backprop --lr 0.001

python  run_attack.py --cuda --method grad_times_input--lr 0.001

respectively

**Comment by pankessel** [comment]
I see. In this case, the network uses relu activations, see

https://github.com/pankessel/explanations_can_be_manipulated/blob/master/src/run_attack.py#L46

As explained in the paper, the vanishing second derivative of relu leads to problems when constructing the adversarial perturbation. You can either switch on beta growth or set a fixed value of beta in the line linked above.

I hope this helps. Please let us know if you have any further questions.

**Comment by laura-rieger** [comment]
My bad, I misread this line and thought it uses a fixed beta if beta growth is *not* specified. I see what you mean. Thanks for your help
